airbyte-integrations/connectors/source-visma-economic/source_visma_economic/source.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple, Union
+from urllib.parse import urlparse, parse_qs
+
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream, HttpSubStream
+
+
+class VismaEconomicStream(HttpStream, ABC):
+    url_base: str = "https://restapi.e-conomic.com/"
+    page_size: int = 1000
+
+    def __init__(self, app_secret_token: str = None, agreement_grant_token: str = None):
+        self.app_secret_token: str = app_secret_token
+        self.agreement_grant_token: str = agreement_grant_token
+        super().__init__()
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        response_json = response.json()
+        if 'pagination' in response_json:
+            if 'nextPage' in response_json['pagination']:
+                parsed_url = urlparse(response_json['pagination']['nextPage'])
+                query_params = parse_qs(parsed_url.query)
+                return query_params
```

Comment:
```suggestion
        if 'nextPage' in response_json.get('pagination').keys():
            parsed_url = urlparse(response_json['pagination']['nextPage'])
            query_params = parse_qs(parsed_url.query)
             return query_params
```

Reply:
You can simplify the code. Please add one unit test to make sure this is working.